### ISSUE TITLE
require.NoError does not work for given example

### ISSUE THREAD

**Issue by MaerF0x0** [stretchr/testify#762]
viewing https://godoc.org/github.com/stretchr/testify/require#Assertions.NoError 

gives the example 
```golang
actualObj, err := SomeFunction()
  if a.NoError(err) {
	   assert.Equal(t, expectedObj, actualObj)
  }
```

but require.NoError does not return a value .  

IMO it should return a bool to match assert.NoError 

let me know if you want a PR to fix.

**Comment by temoto** [comment]
@MaerF0x0 basically `require.X = if !assert.X(...) { t.FailNow() }` which ends test. It can't possibly return anything. I was confused about async documentation at first. You can easily tell it's just an unfortunate implication of `require` code being fully generated from `assert` code, including comments.

**Comment by beono** [comment]
Agree. This confuses a bit, 'till you understand that the whole file is copied from assert package.

**Comment by xordspar0** [comment]
Is there currently a way to change comments in the generated `require` package?

**Comment by beono** [comment]
> Is there currently a way to change comments in the generated `require` package?

As I understand the only way to change this is to modify the template:
https://github.com/stretchr/testify/edit/master/require/require.go.tmpl
```
+ // {{.DocInfo.Name}} is an autogenerated wrapper over assert.{{.DocInfo.Name}}
+ // The signature and examples may not match:
+ //
  {{.Comment}}
  func {{.DocInfo.Name}}(t TestingT, {{.Params}}) {
  	if h, ok := t.(tHelper); ok { h.Helper() }
  	if assert.{{.DocInfo.Name}}(t, {{.ForwardedParams}}) { return }
  	t.FailNow()
  }
```
Then it will produce:
```go
// Error is an autogenerated wrapper over assert.Error.
// The signature and examples may not match:
//
// Error asserts that a function returned an error (i.e. not `nil`).
//
//   actualObj, err := SomeFunction()
//   if assert.Error(t, err) {
// 	   assert.Equal(t, expectedError, err)
//   }
func Error(t TestingT, err error, msgAndArgs ...interface{}) {
	if h, ok := t.(tHelper); ok {
		h.Helper()
	}
	if assert.Error(t, err, msgAndArgs...) {
		return
	}
	t.FailNow()
}
```

**Comment by xordspar0** [comment]
What if the template didn't include the original comment, but instead had the part about it being a wrapper followed by a link to the original in the assert package? I'm not sure if that's more or less user friendly.

**Comment by beono** [comment]
Well, I guess a link in comments is not a convenient way to read documentation in some cases.

**Comment by temoto** [comment]
Please don't make linky docs.
